data/abilities.ts

```diff
@@ -1860,6 +1860,7 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		},
 		onEnd(pokemon) {
 			if (pokemon.illusion) {
+				pokemon.apparentType = pokemon.illusion.apparentType;
```

Comment:
An alternative is this:
```suggestion
				if (!pokemon.terastallized) {
					this.add('-end', pokemon, 'typechange', '[silent]');
					pokemon.apparentType = pokemon.baseSpecies.types.join('/');
				}
```
This fixes the issue in the OP but causes actual type changes (Soak) to disappear until the end of the turn. Maybe that's better, I can't decide.